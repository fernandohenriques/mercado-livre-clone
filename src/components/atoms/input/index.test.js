import React from 'react';
import { mount, render as renderDOM } from 'enzyme';
import { render } from '@testing-library/react';
import Input from './index';

describe('<Input />', () => {
  it('Snapshot testing', () => {
    const { asFragment } = render(
      <Input type="text" placeholder="Search products..." autofocus />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('Should use props className as CSS class', () => {
    const wrapper = renderDOM(
      <Input type="text" className="test-input" />
    );

    expect(wrapper.hasClass('input')).toBe(true);
    expect(wrapper.hasClass('test-input')).toBe(true);
  });

  it('Should props autofocus as default false', () => {
    const wrapper = mount(
      <Input type="text" placeholder="Enter text here..." />
    );

    expect(wrapper.props().autofocus).toBe(false);
  });
});
