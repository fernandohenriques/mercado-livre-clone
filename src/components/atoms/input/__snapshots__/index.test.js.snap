// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Snapshot testing 1`] = `
<DocumentFragment>
  <input
    class="input"
    placeholder="Search products..."
    type="text"
  />
</DocumentFragment>
`;
